python/ray/dag/context.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+import os
+import threading
+from typing import Optional
+from ray.util.annotations import DeveloperAPI
+
+# The context singleton on this process.
+_default_context: "Optional[DAGContext]" = None
+_context_lock = threading.Lock()
+
+# Default buffer size is 1MB.
+DEFAULT_BUFFER_SIZE_BYTES = int(os.environ.get("RAY_DAG_buffer_size_bytes", 1e6))
+# Default async_max_queue_size is 0, which means no limit.
+DEFAULT_ASYNC_MAX_QUEUE_SIZE = int(os.environ.get("RAY_DAG_async_max_queue_size", 0))
+# The default max_buffered_results is 1000, and the default buffer size is 1 MB.
+# The maximum memory usage for buffered results is 1 GB.
+DEFAULT_MAX_BUFFERED_RESULTS = int(os.environ.get("RAY_DAG_max_buffered_results", 1000))
```

Comment:
Remove `DEFAULT` in the names since these are actually configurable from env

Reply:
I think `DEFAULT` seems to be better. If users don't pass the variables into `experimental_compile`, we use the default values instead. I think it makes sense to me that the default values are configurable.